spec/lib/bolt_spec/plans/mock_executor.rb

```diff
@@ -0,0 +1,196 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+require 'bolt/result_set'
+require 'bolt/result'
+require 'set'
+
+module BoltSpec
+  module Plans
+    class UnexpectedInvocation < ArgumentError; end
+
+    # Nothiing in the TaskDouble is 'public'
+    class TaskDouble
+      def initialize
+        @stubs = []
+      end
+
+      def process(targets, task, arguments, options)
+        # TODO: should we bother matching at all? or just call each
+        # stub until one works?
```

Comment:
There is no way to pass a block to stubs at this point. My thought was that in general users would want to pass a single block for both matching and computing the return value in most cases so if it had side effects it might get called twice.